sdcm/sct_provision/region_definition_builder.py

```diff
@@ -81,14 +81,16 @@ def build_instance_definition(self, region: str, node_type: NodeTypeType, index:
                               "NodeIndex": str(index)}
         user_data = self._get_user_data_objects(node_type=node_type, instance_name=name)
         mapper = self.SCT_PARAM_MAPPER[node_type]
+        use_public_ip = self.params.get("ip_ssh_connections") == "public" or node_type == "monitor"
```

Comment:
`ip_ssh_connections` will be deprecated. It's better to call `sdcm.provision.network_configuration.ssh_connection_ip_type` function

Reply:
`sdcm.provision.network_configuration.ssh_connection_ip_type` takes into account the fact that not every backend support new configuration. And use `ip_ssh_connections` if new network config is not supported yet. 
```
    if scylla_network_config := params.get("scylla_network_config"):
        ssh_ip_type = [conf for conf in scylla_network_config if conf["address"] == "test_communication"][0]
        match ssh_ip_type['ip_type']:
            case "ipv6":
                return "ipv6"
            case "ipv4":
                return "public" if ssh_ip_type["public"] else "private"

    # TODO: Keep the "ip_ssh_connections" parameter for non-AWS clusters where new network configuration is not used
    return params.get("ip_ssh_connections") or "private"
```

So you can use `sdcm.provision.network_configuration.ssh_connection_ip_type` function. Once it will be supported, you won't need to replace this string,  it will continue to work.
```
use_public_ip = ssh_connection_ip_type(self.params) or node_type == "monitor"`
```

Reply:
ok, I see, I've forgotten about this function. Used it.
But I'd leave ip_ssh_connections param in unit tests to the time we support the new config.